UKHO.PeriodicOutputService/UKHO.BESS.BuilderService/Program.cs

```diff
@@ -112,27 +114,42 @@ private static HostBuilder BuildHostConfiguration()
+                 if (configuration != null)
                  {
-                     services.AddSingleton<IConfiguration>(configurationBuilder);
-                     services.Configure<BessStorageConfiguration>(configurationBuilder.GetSection("BessStorageConfiguration"));
-                     services.Configure<EssManagedIdentityConfiguration>(configurationBuilder.GetSection("ESSManagedIdentityConfiguration"));
-                     services.Configure<FssApiConfiguration>(configurationBuilder.GetSection("FSSApiConfiguration"));
-                     services.Configure<EssApiConfiguration>(configurationBuilder.GetSection("ESSApiConfiguration"));
-
-                     services.AddDistributedMemoryCache();
-
-                     services.AddSingleton<IAuthFssTokenProvider, AuthTokenProvider>();
-                     services.AddSingleton<IAuthEssTokenProvider, AuthTokenProvider>();
-                     services.AddScoped<IBuilderService, Services.BuilderService>();
-                     services.AddScoped<IEssService, EssService>();
-                     services.AddHttpClient();
-                     services.AddTransient<IEssApiClient, EssApiClient>();
-                     services.AddTransient<IFssApiClient, FssApiClient>();
+                     serviceCollection.AddSingleton(configuration);
+                     serviceCollection.Configure<BessStorageConfiguration>(configuration.GetSection("BessStorageConfiguration"));
+                     serviceCollection.Configure<EssManagedIdentityConfiguration>(configuration.GetSection("ESSManagedIdentityConfiguration"));
+                     serviceCollection.Configure<FssApiConfiguration>(configuration.GetSection("FSSApiConfiguration"));
+                     serviceCollection.Configure<EssApiConfiguration>(configuration.GetSection("ESSApiConfiguration"));
+                     configuration.Bind("FSSApiConfiguration", fssApiConfiguration);
                  }
+
+                 serviceCollection.AddDistributedMemoryCache();
+
+                 serviceCollection.AddSingleton<IAuthFssTokenProvider, AuthTokenProvider>();
+                 serviceCollection.AddSingleton<IAuthEssTokenProvider, AuthTokenProvider>();
+                 serviceCollection.AddScoped<IBuilderService, Services.BuilderService>();
+                 serviceCollection.AddScoped<IEssService, EssService>();
+                 serviceCollection.AddScoped<IFssService, FssService>();
+                 serviceCollection.AddHttpClient();
+                 serviceCollection.AddTransient<IEssApiClient, EssApiClient>();
+                 serviceCollection.AddTransient<IFssApiClient, FssApiClient>();
+
+                 serviceCollection.AddScoped<IFileSystemHelper, FileSystemHelper>();
+                 serviceCollection.AddScoped<IFileSystem, FileSystem>();
+                 serviceCollection.AddScoped<IZipHelper, ZipHelper>();
+                 serviceCollection.AddScoped<IFileUtility, FileUtility>();
+
```

Comment:
wouldn't this be neater to remove the clutter and abstract

Reply:
Hi Ray, this is a standard service collection setup format in program.cs, no change required here